apps/calendar/js/app.js

```diff
@@ -23,225 +24,185 @@ var storeFactory = require('store/factory');
+    // as soon as possible
+    var storeFactory = core.storeFactory;
+    var calendars = storeFactory.get('Calendar');
+    var startDayObserver = calendars.all().then(() => dayObserver.init());
 
-    notificationsController.observe();
-    periodicSyncController.observe();
-
-    var recurringEventsController = new RecurringEventsController();
-    this.observePendingObject(recurringEventsController);
-    recurringEventsController.observe();
-    this.recurringEventsController = recurringEventsController;
-
-    // turn on the auto queue this means that when
-    // alarms are added to the database we manage them
-    // transparently. Defaults to off for tests.
-    core.storeFactory.get('Alarm').autoQueue = true;
-  },
-
-  _initUI: function() {
-    // re-localize dates on screen
-    dateL10n.init();
-
-    timeObserver.init();
-    core.timeController.move(new Date());
-
-    viewFactory.get('TimeHeader', header => header.render());
-    viewFactory.get('ViewSelector', tabs => tabs.render());
-
-    document.body.classList.remove('loading');
-
-    // at this point we remove the .loading class and user will see the main
-    // app frame
-    performance.domLoaded();
-
-    this._routes();
-
-    nextTick(() => viewFactory.get('Errors'));
-
-    // Restart the calendar when the timezone changes.
-    // We do this on a timer because this event may fire
-    // many times. Refreshing the url of the calendar frequently
-    // can result in crashes so we attempt to do this only after
-    // the user has completed their selection.
-    window.addEventListener('moztimechange', () => {
-      debug('Noticed timezone change!');
-      // for info on why we need to restart the app when the time changes see:
-      // https://bugzilla.mozilla.org/show_bug.cgi?id=1093016#c9
-      nextTick(this.forceRestart);
+    // we init the UI after the db.load to increase perceived performance
+    // (will feel like busytimes are displayed faster)
+    var startUI = new Promise(resolve => {
+      navigator.mozL10n.once(() => setupUI().then(resolve));
     });
-  },
-
-  /**
-   * Primary code for app can go here.
-   */
-  init: function() {
-    debug('Will initialize calendar app...');
-
-    this.forceRestart = this.forceRestart.bind(this);
 
-    if (!core.db) {
-      this.configure();
-    }
-
-    core.db.load(() => {
-      this._initControllers();
-      // it should only start listening for month change after we have the
-      // calendars data, otherwise we might display events from calendars that
-      // are not visible. this also makes sure we load the calendars as soon as
-      // possible
-      core.storeFactory.get('Calendar').all(() => dayObserver.init());
-
-      // we init the UI after the db.load to increase perceived performance
-      // (will feel like busytimes are displayed faster)
-      navigator.mozL10n.once(() => this._initUI());
+    return Promise.all([startDayObserver, startUI]).then(() => {
```

Comment:
nit: maybe create 2 functions `startDayObserver()` and `startUI()`? (or maybe make just make sure `setupUI` waits for the `navigator.mozL10n.once` and returns a promise?)